nixos/modules/services/web-apps/alist.nix

```diff
@@ -0,0 +1,307 @@
+                example = "127.0.0.1";
+                description = "database host";
+              };
+              port = lib.mkOption {
+                type = lib.types.port;
+                default = 0;
+                example = 3306;
+                description = "database port";
+              };
+              user = lib.mkOption {
+                type = lib.types.str;
+                default = "";
+                example = "root";
+                description = "database account";
+              };
+              password = lib.mkOption {
+                type = lib.types.oneOf [
+                  lib.types.str
+                  (lib.types.attrsOf lib.types.path)
+                ];
+                default = "";
+                example = lib.literalExpression ''
+                  {
+                    _secret = /run/agenix/alist-db-password;
+                  };
+                '';
+                description = "database password";
+              };
+              name = lib.mkOption {
+                type = lib.types.str;
+                default = "";
+                example = "alist";
+                description = "database name";
+              };
+              db_file = lib.mkOption {
+                type = lib.types.str;
+                default = "/var/lib/alist/data/data.db";
+                description = "Database location, used by sqlite3";
+              };
+              table_prefix = lib.mkOption {
+                type = lib.types.str;
+                default = "x_";
+                description = "database table name prefix";
+              };
+              ssl_mode = lib.mkOption {
+                type = lib.types.str;
+                default = "";
+                description = ''
+                  To control the encryption options during the SSL handshake, the parameters can be searched by themselves, or check the answer from ChatGPT on https://alist.nn.ci/config/configuration.html#database
+                '';
+              };
+              dsn = lib.mkOption {
+                type = lib.types.str;
+                default = "";
+                description = ''
+                  see https://github.com/alist-org/alist/pull/6031
+                '';
+              };
+            };
+            scheme = {
+              address = lib.mkOption {
+                type = lib.types.str;
+                default = "0.0.0.0";
+                description = ''
+                  The http/https address to listen on, default `0.0.0.0`
+                '';
+              };
+              http_port = lib.mkOption {
+                type = lib.types.ints.between (-1) 65535;
+                default = 5244;
+                description = ''
+                  The http port to listen on, default `5244`, if you want to disable http, set it to `-1`
+                '';
+              };
+              https_port = lib.mkOption {
+                type = lib.types.ints.between (-1) 65535;
```

Comment:
Is 0 allowed?

We should probably use the enum type with -1 and the ports type.